### DISCUSSION TITLE
Rename`attributesToRetrieve` from the documents API.

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#440] [Feedback & Feature Proposal]
## Why?

**The documents API is used to manage documents within an index.**

[Since we are touching the documents API to remove the fact `displayeAttributes` setting impacts this API](https://github.com/meilisearch/product/discussions/432), I think we should also rename `attributesToRetrieve` on this API and use a generic name that does not collide with the search endpoints.

I would propose something like `fieldset` e.g. `fieldset=title,name,price` which would work exactly like `attributesToRetrieve` today on the documents API but could be used generically on other APIs like `Task`, `Indexes`, `API Keys` etc in the future. It would not be bounded to `attributesToRetrieve` which is kept on the `search` endpoints.

In this way, we will have endpoints dedicated to resource API management consistent with each other, while search uses its own, more contextualized term akka `attributesToRetrieve`

**Comment by curquiza** [comment]
Why not simply `fields`?

**Reply by gmourier** [reply]
"Boom, easy as that" 😎

**Reply by bidoubiwa** [reply]
I agree 💯

**Comment by curquiza** [comment]
So `fields` it is! 🚀

**Comment by gmourier** [comment]
Following the [v0.28 release](https://github.com/meilisearch/meilisearch/releases/tag/v0.28.0), this discussion is now locked.

We encourage you to create a new thread if needed.

Thank you ✨
